Sources/HttpPipeline/SignedCookies.swift

```diff
@@ -0,0 +1,148 @@
+import CommonCrypto
+import Cryptor
+import Foundation
+
+extension ResponseHeader {
+  /// Computes a signed cookie.
+  ///
+  /// - Parameters:
+  ///   - key: The name of the cookie.
+  ///   - value: The data of the cookie to sign.
+  ///   - options: (Optional) Extra options to attach to the cookie.
+  ///   - secret: The secret to sign the cookie with. This value must also be provided to verify a signature.
+  ///   - encrypt: (Optional) Further encrypts the signed cookie using the secret provided.
+  /// - Returns: A `Set-Cookie` header containing the signed cookie.
+  public static func setSignedCookie(
+    key: String,
+    data: Data,
+    options: Set<CookieOption> = [],
+    secret: String,
+    encrypt: Bool = false
+    )
+    -> ResponseHeader? {
```

Comment:
I had to make the decision of whether or not the return value should be optional. there are a lot of things that can go wrong along the way, and so it seemed weird to coalesce to a non-`nil` value at the end.

we may even want to use `Either` have a proper error enum